File: Maximum Unique Subarray Sum After Deletion/Maximum Unique Subarray Sum After Deletion.java
// Solution for Maximum Unique Subarray Sum After Deletion in JAVA

class Solution {
    public int maxSum(int[] nums) {
        int[] seen = new int[101];
        Arrays.fill(seen, -1);
        int maxSum = 0;
        int maxNegative = Integer.MIN_VALUE;

        for (int num : nums) {
            if (num <= 0) {
                maxNegative = Math.max(maxNegative, num);
            } else if (seen[num] == -1) {
                maxSum += num;
                seen[num] = 1;
            }
        }

        return maxSum == 0 ? maxNegative : maxSum;
    }
}

File: Maximum Unique Subarray Sum After Deletion/Maximum Unique Subarray Sum After Deletion.py
# Solution for Maximum Unique Subarray Sum After Deletion in PY

class Solution:
    def maxSum(self, nums: List[int]) -> int:
        seen = [-1] * 101
        maxSum = 0
        maxNegative = float('-inf')

        for num in nums:
            if num <= 0:
                maxNegative = max(maxNegative, num)
            elif seen[num] == -1:
                maxSum += num
                seen[num] = 1

        return maxNegative if maxSum == 0 else maxSum
